app/src/main/java/com/spascoding/taskycourse/feature_register_screen/util/AuthPattern.kt

```diff
@@ -0,0 +1,31 @@
+package com.spascoding.taskycourse.feature_register_screen.util
+
+import android.util.Patterns
+
+object AuthPattern {
+    fun NAME(text: String): Boolean = text.length in 4..50
+    fun EMAIL(text: String): Boolean = Patterns.EMAIL_ADDRESS.matcher(text).matches()
+    fun PASSWORD(text: String): Boolean {
+        // Check if the length is at least 9 characters
+        if (text.length < 9) return false
+
+        // Initialize flags for each condition
+        var hasLowercase = false
+        var hasUppercase = false
+        var hasDigit = false
+
+        // Iterate through each character in the text
+        for (char in text) {
+            when {
+                char.isLowerCase() -> hasLowercase = true
+                char.isUpperCase() -> hasUppercase = true
+                char.isDigit() -> hasDigit = true
+            }
+            // If all conditions are met, return true
+            if (hasLowercase && hasUppercase && hasDigit) return true
+        }
+
+        // Return false if any condition is not met
+        return false
```

Comment:
I'd recommend to play around more with the list functions from the Kotlin standard lib. You can achieve the same in a lot more readable way like this: 
```
val hasLowerCase = password.any { it.isLowerCase() }
val hasUpperCase = password.any { it.isUpperCase() }
val hasDigit = password.any { it.hasDigit() }
val hasValidLength = password.length in min..max

return hasLowerCase && hasUpperCase && hasDigit && hasValidLength
```